### PR TITLE
Replication type

### PR DESCRIPTION
**Pull request by raphink**
This provides a type to setup replication slots on a Postgresql master server.

### CODE REVIEW

**Comment by tphoney** [comment]
Thanks for the work, this needs a little bit of cleanup, Could you add some documentation then make sure the tests pass and squash your commits.

**Comment by raphink** [comment]
@tphoney I fixed the tests & squashed the commits.

The existing types for this module are not documented in README.md. They'll get their doc in the types tab on the forge anyway.

File: lib/puppet/type/postgresql_replication_slot.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Puppet::Type.newtype(:postgresql_replication_slot) do
+  ensurable
```
Comment (hunner) [inline lib/puppet/type/postgresql_replication_slot.rb]: Could you add an `@doc` description, similar to https://github.com/puppetlabs/puppetlabs-mysql/blob/master/lib/puppet/type/mysql_user.rb#L3 (well, more verbose than that would be great).

**Comment by raphink** [comment]
@hunner done

**Pull request merged**